tests/unit/src/main/scala/tests/TestingClient.scala

```diff
@@ -66,6 +69,24 @@ final class TestingClient(workspace: AbsolutePath, buffers: Buffers)
   ): Unit = {
     clientCommands.addLast(params)
   }
+
+  override def applyEdit(
+      params: ApplyWorkspaceEditParams
+  ): CompletableFuture[ApplyWorkspaceEditResponse] = {
+    def applyEdits(uri: String, textEdits: java.util.List[TextEdit]): Unit = {
+      val path = AbsolutePath(uri)
+
```

Comment:
I would lose the additional newlines here too.

Reply:
newlines are great. And here they separate the three steps of the function quite well:
1. prepare the input
2. apply the edit
3. write the new content

maybe 1. and 2. could be aggregated into?
```scala
      val content = uri.toAbsolutePath.readText
      val editedContent = 
        TextEdits.applyEdits(content, textEdits.asScala.toList)
```

Reply:
It doesn't work.
`Illegal character in path at index 44: /home/tdudzik/Projects/metals/target/e2e/add package/single-level/a/src/main/scala/a/Main.scala` 

Also, we are using `AbsolutePath` two times (`readText` and `writeText`) so I think it's good idea to have the `path` value.

Reply:
Illegal path seems to be because of the whitespace I think.

Reply:
Yes but after fixing whitespace issue another exception is thrown `Missing scheme
java.lang.IllegalArgumentException: Missing scheme`

Reply:
Ok, let's jsut leave whatever works, that's jsut the testing client